tests/test_port_dpb_neigh.py

```diff
@@ -0,0 +1,241 @@
+from swsscommon import swsscommon
+import redis
+import time
+import os
+import pytest
+import json
+import re
+from port_dpb import DPB
+
+@pytest.mark.usefixtures('dpb_setup_fixture')
+class TestPortDPBNeigh(object):
+    def clear_srv_config(self, dvs):
+        dvs.servers[0].runcmd("ip address flush dev eth0")
+        dvs.servers[1].runcmd("ip address flush dev eth0")
+
+    def test_set_neighbor_DPB(self, dvs, testlog):
+        dvs.setup_db()
+
+        # bring up interface
+        dvs.set_interface_status("Ethernet0", "up")
+
+        # create interface and get rif_oid
+        rif_oid = dvs.create_l3_intf("Ethernet0", "")
+
+        # assign IP to interface
+        dvs.add_ip_address("Ethernet0", "10.0.0.1/24")
+
+        # add neighbor
+        dvs.set_neighbor("Ethernet0", "10.0.0.2", "00:01:02:03:04:05")
+
+        # check application database
+        tbl = swsscommon.Table(dvs.pdb, "NEIGH_TABLE:Ethernet0")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 1
+        assert intf_entries[0] == "10.0.0.2"
+        (status, fvs) = tbl.get(intf_entries[0])
+        assert status == True
+        assert len(fvs) == 2
+        for fv in fvs:
+            if fv[0] == "neigh":
+                assert fv[1] == "00:01:02:03:04:05"
+            elif fv[0] == "family":
+                assert fv[1] == "IPv4"
+            else:
+                assert False
+
+        # check ASIC neighbor database
+        tbl = swsscommon.Table(dvs.adb, "ASIC_STATE:SAI_OBJECT_TYPE_NEIGHBOR_ENTRY")
+        intf_entries = tbl.getKeys()
+        assert len(intf_entries) == 1
+        route = json.loads(intf_entries[0])
+        assert route["ip"] == "10.0.0.2"
+        #assert route["rif"] == rif_oid
+
+        (status, fvs) = tbl.get(intf_entries[0])
+        assert status == True
+        for fv in fvs:
```

Comment:
I would prefer to convert fvs into dictionary and then assert fvs_dict["SAI_NEIGHBOR_ENTRY_ATTR_DST_MAC_ADDRESS"] == "00:01:02:03:04:05".

That's because if we dont have SAI_NEIGHBOR_ENTRY_ATTR_DST_MAC_ADDRESS in fvs, this would claim the test case as pass. Which we dont want.